HeterogeneousCore/SonicCore/src/RetrySameServerAction.cc

```diff
@@ -0,0 +1,16 @@
+#include "HeterogeneousCore/SonicCore/interface/RetrySameServerAction.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicClientBase.h"
+
+void RetrySameServerAction::retry() {
+  ++tries_;
+  //if max retries has not been exceeded, call evaluate again
+  if (tries_ < allowedTries_) {
+    eval();
+    return;
+  } else {
+    shouldRetry_ = false;  // Flip flag when max retries are reached
+    edm::LogInfo("RetrySameServerAction") << "Max retry attempts reached. No further retries.";
+  }
+}
+
+DEFINE_EDM_PLUGIN(RetryActionFactory, RetrySameServerAction, "RetrySameServerAction");
```

Comment:
it may be worth defining another macro for this:
```cpp
#define DEFINE_RETRY_ACTION(type) \
  DEFINE_EDM_PLUGIN(RetryActionFactory, type, #type);
#endif
```

Reply:
done